Module2_HW2/Module2_HW2/Store.cs

```diff
@@ -0,0 +1,239 @@
+                if (_listProducts[i].Name == product.Name)
+                {
+                    return i;
+                }
+            }
+            return -100;
+        }
+        public void AddToStorage(Product product)
+        {
+            int index = FindIndexInStorage(product);
+            if (index >= 0)
+            {
+                _listProducts[index].Amount = _listProducts[index].Amount + product.Amount;
+            }
+            else 
+            {
+                _listProducts.Add(product);
+            }
+        }
+
+        public bool RemoveFromStorage(Product product)
+        {
+            int index = FindIndexInStorage(product);
+            if (index >= 0)
+            {
+                if (_listProducts[index].Amount < product.Amount)
+                {
+                    Console.WriteLine("You can't remove this product. Please  review store");
+                    return false;
+                }
+                _listProducts[index].Amount = _listProducts[index].Amount - product.Amount;
+                return true;
+            }
+            else
+            {
+                Console.WriteLine("You can't remove this product. Please  review store");
+                return false;
+            }
+        }
+
+        public bool ReviewStore() 
+        {
+            if (_listProducts.Count == 0) 
+            {
+                return false;
+            }
+            foreach (var item in _listProducts) 
+            {
+                Console.WriteLine($"Name {item.Name}, Amount {item.Amount}");
+            }
+            return true;
+        }
+
+        public bool AddToCart(Product products)
+        {
+            bool flag =_shoppingCart.AddToCart(products);
+            return flag;
+        }
+
+        public bool RemoveFromCart(Product products)
+        {
+            bool flag = _shoppingCart.RemoveFromCart(products);
+            return flag;
+        }
+
+        public void CartConfirmation()
+        {
+            _shoppingCart.CartConfirmation();
+        }
+
+        public void ReviewShopingCart()
+        {
+            _shoppingCart.DisplayShopingCart();
+        }
+
+        private void AddDataToStorage()
+        {
+            Product product1 = new Product("keyboard", 10);
+            Product product2 = new Product("TV", 20);
+            Product product3 = new Product("display", 30);
+            Product product4 = new Product("PC", 5);
+            Product product5 = new Product("DVD", 12);
+            Product product6 = new Product("CD", 50);
+            Product product7 = new Product("Microphone", 15);
+
+            AddToStorage(product1);
+            AddToStorage(product2);
+            AddToStorage(product3);
+            AddToStorage(product4);
+            AddToStorage(product5);
+            AddToStorage(product6);
+            AddToStorage(product7);
+        }
+
+        private class ShopingCart
+        {
+            private Store _store;
```

Comment:
Do not see where this field is initialized. How is it used?

Reply:
I use _store for calling methods from public class Store